core/src/main/java/be/xplore/pricescraper/services/ItemServiceImpl.java

```diff
@@ -127,7 +127,7 @@ private void storeItemPrice(TrackedItem item, double price) {
   public ServiceResponse<TrackedItem> addTrackedItem(String urlToItem) {
```

Comment:
service response object seems a bit weird to me

you could return the TrackedItem and in any of the failed cases you could throw an exception and use @ExceptionHandler methods to translate these into the correct statuscode / message

results in clearer code